File: packages/core/python/addons/manifest.py
from typing import List, Optional, Union
from pydantic import BaseModel, Field


# Base models for Catalog 'extra' options
class ExtraOption(BaseModel):
    name: str
    is_required: bool = Field(False, alias="isRequired")
    options: Optional[List[str]] = None
    options_limit: Optional[int] = Field(None, alias="optionsLimit")


# Base model for Catalog definitions
class Catalog(BaseModel):
    id: str
    type: str
    name: str
    extra: Optional[List[ExtraOption]] = None


# Base model for Resource definitions
class Resource(BaseModel):
    name: str
    types: List[str]
    id_prefixes: Optional[List[str]] = Field(None, alias="idPrefixes")


# The main AddonManifest model
class AddonManifest(BaseModel):
    id: str
    version: str
    name: str
    description: str
    resources: List[Union[str, Resource]]
    types: List[str]

    logo: Optional[str] = None
    background: Optional[str] = None
    catalogs: List[Catalog] = []
    id_prefixes: Optional[List[str]] = Field(None, alias="idPrefixes")

    class Config:
        populate_by_name = True  # Allows using alias like 'isRequired'

File: packages/core/typescript/addons/manifest.ts
export interface ExtraOption {
  name: string;
  isRequired?: boolean;
  options?: string[];
  optionsLimit?: number;
}

export interface Catalog {
  id: string;
  type: string;
  name: string;
  extra?: ExtraOption[];
}

export interface Resource {
  name: string;
  types: string[];
  idPrefixes?: string[];
}

export interface AddonManifest {
  id: string;
  version: string;
  name: string;
  description: string;
  resources: (string | Resource)[];
  types: string[];

  logo?: string;
  background?: string;
  catalogs: Catalog[];
  idPrefixes?: string[];
}